pkg/server/container_stop.go

```diff
@@ -94,11 +93,16 @@ func (c *criContainerdService) stopContainer(ctx context.Context, container cont
 			}
 		}
 		glog.V(2).Infof("Stop container %q with signal %v", id, stopSignal)
-		_, err = c.taskService.Kill(ctx, &tasks.KillRequest{
-			ContainerID: id,
-			Signal:      uint32(stopSignal),
-			All:         true,
-		})
+
+		task, err := container.Container.Task(ctx, nil)
+		if err != nil {
+			if !isContainerdGRPCNotFoundError(err) {
+				return fmt.Errorf("failed to stop container, task not found for container %q: %v", id, err)
+			}
```

Comment:
And we could get task outside of the if clause, so that we don't need to do it twice.